### ISSUE TITLE
Multi-gpu Training Question

### ISSUE THREAD

**Issue by NayeeC** [NKI-AI/direct#237]
Hi!
I run the code in Linux and try to run with `num-gpus=2`, but it doesn't seem to shorten the running time compared with `num-gpus=1`. How can I use multi-GPUs to accelerate the training?

**Comment by georgeyiasemis** [comment]
Hi @NayeeC.

Using multiple gpus shouldn't reduce training time. You can adapt training setting in the configuration `.yaml` file under `training`.
For example you can change `<num_iterations>`:

```
training:
  datasets:
    ...
  num_iterations: <num_iterations>
  validation_steps: <num_val_steps>
  ...
```

**Comment by georgeyiasemis** [comment]
Hi @NayeeC, to explain a bit further:

In the case of DIRECT, using multiple GPUs can be seen as increasing the batch size. This is because when using multiple GPUs, the training data is split and distributed across multiple GPUs. The models on each GPU then perform forward and backward computations in parallel, effectively increasing the batch size.

The implementation of multi-GPU training in DIRECT uses the PyTorch torch.distributed package, which handles the distribution of data and models across GPUs, as well as communication between GPUs during the forward and backward computations.

**Comment by NayeeC** [comment]
Hi @georgeyiasemis 
Thank you for getting back to me.
I ran into another question when I end up training with a piece of unexpected information: `terminate called without an active exception`  in Linux with multi-GPUs. But this information didn't appear in Windows.
Could you tell me:
1. Whether this affect the experiment or not;
2. How can I get rid of it?
Thank you in advance.

**Comment by georgeyiasemis** [comment]
Hi @NayeeC. I cannot answer that based on only that. 

For further investigation, please:
* Open a new issue
* Describe the bug
 and provide a clear and concise description of what the bug is.
* Report the error trace
* Report the versions of DIRECT, pytorch, cuda, etc.
